### DISCUSSION TITLE
Catapulta, deployments tool compatible with ZkSync

### DISCUSSION THREAD

**Discussion by kartojal** [zkSync-Community-Hub/zksync-developers#142] [Show and Tell]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

Hello there! wanted to share a deployments tool compatible with ZkSync named Catapulta, currently in alpha, and I am looking for :zksync: ZkSync solidity developers :zksync:  to test the tool for free during alpha!

:link:  https://catapulta.sh/docs  

Catapulta allows you to:
- Automate verifications in ZkSync main block explorer
- Generate deployment reports in a private dashboard, to keep track of past deployments, artifacts and txs.
- Automate RPC configs in hardhat.config.js/ts

This tool aims to improve the DevX for interacting with Solidity and ZkSync, making it a bit easier to verify smart contracts and keep track of deployments.

Last week made it work with basic ZkSync deployments, but need to test further with more complex deployment scenarios to improve the ZkSync support.

For using Catapulta in a ZkSync project, you need to:
1. Sign up in Catapulta dashboard at https://catapulta.sh  and create a Project, all production usage is free during alpha.
2. Install Catapulta CLI  in your Hardhat project
```
npm i catapulta
```
3. At the Catapulta plugin at hardhat.config.ts/js:
```
// At hardhat.config.ts/js
require('catapulta/hardhat');
```
4. Copy the `CATAPULTA_API_KEY` into your .env file
5. Them proceed to run your deployments with deploy-zksync as always, with catapulta:
```
npx catapulta deploy-zksync --network zksyncTestnet
```
if you use `yarn, you would be able to run via:
```
yarn catapulta deploy-zksync --network zksyncTestnet 
```

Once the deploy-zksync finished, the contracts should be verified automatically without any extra stuff.

thanks @bxpana  for allowing me to share this.

### What is the specific issue or error you're encountering?

N/A

### Can you share the error messages or logs you're receiving, if any?

N/A

### Have you made any recent changes to the contract before encountering this issue?

N/A

### Are there any external libraries or contracts that your contract interacts with?

N/A

### Can you provide the relevant portions of your contract code where the issue is occurring?

N/A

### Have you tried to isolate the problem, and if so, what were the results?

N/A

### What steps have you already taken to try to resolve the issue?

N/A

**Comment by caffeinum** [comment]
Looks promising! For some reason, didn't work for me

```shell
bunx catapulta script contracts/AAFactory.sol --chain sepolia

Catapulta.sh 🏏 Forge script deployment (0.1.38)
================================================
Project name: Bold Tortoise
Project URL: https://catapulta.sh/project/650c7c739651675ef9f57d9e
Deployment UUID: 4bf28bde-fc93-4725-8594-362b0d73136b

📀 Building artifacts...

🗜  Compressing artifacts...

📤 Uploading artifacts to the Catapulta DB...

[AxiosError: Request failed with status code 500] {
  code: 'ERR_BAD_RESPONSE',
  config: {
    transitional: {
      silentJSONParsing: true,
      forcedJSONParsing: true,
      clarifyTimeoutError: false
    },
    adapter: [Function: httpAdapter],
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 0,
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: Infinity,
    maxBodyLength: Infinity,
    env: { FormData: [Function] },
    validateStatus: [Function: validateStatus],
    headers: {
      Accept: 'application/json, text/plain, */*',
      'Content-Type': 'multipart/form-data; boundary=--------------------------296594786723014385705612',
      'User-Agent': 'axios/0.27.2',
      'content-type': 'multipart/form-data; boundary=------
...
kBytesWritten)]: 0,
      [Symbol(kNeedDrain)]: true,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype],
      [Symbol(errored)]: null,
      [Symbol(kHighWaterMark)]: 16384,
      [Symbol(kRejectNonStandardBodyWrites)]: false,
      [Symbol(kUniqueHeaders)]: null
    },
    data: { err: 'There was an error uploading the artifacts.' }
  }
}
```

**Reply by kartojal** [reply]
@caffeinum do you have an example repo? thanks!

**Reply by kartojal** [reply]
feel free to DM me at Twitter or Telegram to know more details about your issue, same handle @kartojal

**Comment by johnwesener** [comment]
This deployment tool, Catapulta, seems like a promising addition to the ZkSync ecosystem. Automating verification and deployment processes can significantly streamline the development workflow for ZkSync projects. Solidity developers working with ZkSync should consider testing Catapulta during its alpha phase to provide feedback and help improve its functionality. The steps for integration provided in the description seem straightforward and easy to follow. Thank you for sharing this resource!

**Comment by natyfikapsha** [comment]
Exciting news! Catapulta, an alpha deployment tool compatible with ZkSync, is now available for testing by ZkSync Solidity developers. With Catapulta, you can automate verifications, generate deployment reports, and streamline RPC configurations, enhancing your Solidity-ZkSync development experience. Sign up for free during the alpha phase and get started with Catapulta today! Visit https://catapulta.sh/docs for more information. Happy coding!

**Comment by gauvinjohnny** [comment]
Exciting news for ZkSync developers! Check out Catapulta, an alpha tool designed to streamline deployments and verifications in ZkSync projects. Sign up for free during alpha, automate verifications, generate deployment reports, and simplify RPC configs. Find out more at https://catapulta.sh/docs. Happy coding!

**Comment by moyemv** [comment]
The user is sharing information about a new deployments tool called Catapulta, which is compatible with ZkSync and aims to improve the developer experience for interacting with Solidity and ZkSync. The user is seeking ZkSync Solidity developers to test the tool during alpha and has provided instructions for using Catapulta in a ZkSync project. The user has not reported encountering any specific issue or error and has not provided any relevant code or repository link.

**Comment by Josephtran102** [comment]
Awesome! What I need
